### PR TITLE
ci(integration): add private repo test case

### PR DESCRIPTION
**Pull request by vdice**
**What this PR does / why we need it**:

* Adds a private repo test case to the integration test suite

Close https://github.com/brigadecore/brigade/issues/1233

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: v2/tests/integration_test.go [hunk 1]
```diff
@@ -301,6 +340,12 @@ func TestMain(t *testing.T) {
 	)
 
 	for _, tc := range testcases {
+		// Skip private repo test if required env var not present
+		if tc.name == "GitHub - private repo" &&
+			os.Getenv("BRIGADE_CI_PRIVATE_REPO_SSH_KEY") == "" {
+			continue
+		}
```
Comment (krancour) [inline v2/tests/integration_test.go:347]: Is it worth having a more general mechanism that scales better as we (potentially) continue to add test cases that can _only_ run in CI?

I had been thinking we could add a `bool` field to the `testCase` struct like `skipLocally` or something that is `false` by default, but can be set to `true` for the few cases that can only run in CI.

wdyt?

Comment (krancour) [reply v2/tests/integration_test.go:347]: But consider this non-blocking.

**Review by krancour — APPROVED** [review]

File: v2/tests/integration_test.go (hunk 1 above)
Comment (vdice) [reply v2/tests/integration_test.go:347]: I did consider that, but opted to just wait on adding the toggle until we had more than one test case to utilize it.  Should be quick to add when we get there!

Comment (vdice) [reply v2/tests/integration_test.go:347]: (Actually found it handy to be able to run locally _if_ I had the req'd env var set, for testing 😄 )

**Pull request merged**